fili-core/src/test/groovy/com/yahoo/bard/webservice/web/endpoints/JobsServletReactiveChainforResultsEndpointSpec.groovy

```diff
@@ -152,73 +181,119 @@ class JobsServletReactiveChainforResultsEndpointSpec extends Specification {
 
     def "If the PreResponse is available in the PreResponseStore and we miss the notification from broadcastChannel, we go to the PreResponseStore exactly once"() {
         setup:
-        JobsApiRequest apiRequest1 = new JobsApiRequest(null, "5", "", "", null, uriInfo, jobPayloadBuilder, apiJobStore, mockPreResponseStore, broadcastChannel)
+        JobsApiRequest apiRequest1 = new JobsApiRequest(
+                null,
+                "5",
+                "",
+                "",
+                null,
+                uriInfo,
+                jobPayloadBuilder,
+                apiJobStore
+        )
 
-        when: "PreResponse is available in the PreResponseStore"
-        mockPreResponseStore.save("ticket4", PreResponseTestingUtils.buildPreResponse("2016-04-24")).subscribe()
+        and:
+        PreResponse preResponse = PreResponseTestingUtils.buildPreResponse("2016-04-24")
 
-        and: "We miss the notification that the preResponse is stored in the PreResponseStore"
+        when: "We miss the notification that the preResponse is stored in the PreResponseStore"
```

Comment:
This should also be a `given` piece. The only "when" involved here is the `getResults` call.